Shared/UI/Views/MessageContentView/MessageContentView.swift

```diff
@@ -194,6 +193,35 @@ class MessageContentView: BaseView {
     }
 }
 
+// MARK: - Message Consumption
+
+#if IOS
+extension MessageContentView {
+
+    func configureConsumption(for message: Messageable) {
+        if message.isConsumedByMe {
+            self.textView.setFont(.regular)
+        } else {
+            self.textView.setFont(.regular)
+        }
+    }
+
+    func setToRead() {
+        guard let msg = self.message, msg.canBeConsumed else { return }
+        Task {
+            try await self.message?.setToConsumed()
+        }
+    }
+
+    func setToUnread() {
+        guard let msg = self.message, msg.isConsumedByMe else { return }
+        Task {
+            try await self.message?.setToUnconsumed()
+        }
+    }
+}
+#endif
```

Comment:
Personally I like this in the main file instead of an extension, even if it's target specific. It's just easier for me to find.

Reply:
I agree.